### PR TITLE
ci: Change dotnet formatter

### PR DESCRIPTION
**Pull request by askpt**
## This PR

- This PR changes the code formatter we currently use to the new built-in in the SDK. See https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-format

### Notes
- We should now use this tool since it is now shipped as part of the dotnet SDK, removing the need to install an external one.
- As pointed out in the chat, this build runs in parallel and is relatively fast. I will enable all PRs instead of those focusing on changes to .cs files.

### CODE REVIEW

File: .github/workflows/dotnet-format.yml [hunk 1]
```diff
@@ -2,29 +2,22 @@ name: dotnet format
 
 on:
   push:
-    branches: [ main ]
-    paths:
-    - '**.cs'
-    - '.editorconfig'
```
Comment (beeme1mr) [inline .github/workflows/dotnet-format.yml:8]: Why was this removed?

**Review by beeme1mr — APPROVED** [review]

File: .github/workflows/dotnet-format.yml (hunk 1 above)
Comment (askpt) [reply .github/workflows/dotnet-format.yml:8]: I removed it because this action runs relatively fast and in parallel. If there is any change to the action, we can see if it fails.

Comment (beeme1mr) [reply .github/workflows/dotnet-format.yml:8]: Fair enough, thanks.

**Review by kinyoklion — APPROVED** [review]
